staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go

```diff
@@ -333,10 +371,38 @@ func TestWaitUntilFreshAndList(t *testing.T) {
 	if len(list) != 2 {
 		t.Errorf("unexpected list returned: %#v", list)
 	}
+
+	// list with spec.nodeName index.
+	matchValues = []storage.MatchValue{
+		{IndexName: "l:not-exist-label", Value: "whatever"},
+		{IndexName: "f:spec.nodeName", Value: "node2"},
+	}
+	list, resourceVersion, err = store.WaitUntilFreshAndList(5, matchValues, nil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if resourceVersion != 5 {
+		t.Errorf("unexpected resourceVersion: %v, expected: 5", resourceVersion)
+	}
+	if len(list) != 1 {
+		t.Errorf("unexpected list returned: %#v", list)
+	}
+
+	// list with index not exists.
+	matchValues = []storage.MatchValue{
+		{IndexName: "l:not-exist-label", Value: "whatever"},
+	}
+	list, resourceVersion, err = store.WaitUntilFreshAndList(5, matchValues, nil)
+	if resourceVersion != 5 {
+		t.Errorf("unexpected resourceVersion: %v, expected: 5", resourceVersion)
+	}
+	if len(list) != 3 {
```

Comment:
Hmm.. wait. Why is that 3? Shguldn't it be 0 (see my comment above)?

Reply:
As has no matchValue match, it will fallback to return all data

Reply:
OK - never mind then.